plugins/modules/storagebox_snapshot.py

```diff
@@ -0,0 +1,226 @@
+        "SNAPSHOT_NOT_FOUND": "Snapshot with name {0} does not exist".format(snapshot_name),
+        "SNAPSHOT_LIMIT_EXCEEDED": "Snapshot limit exceeded",
+    }
+    module.fail_json(msg=error_messages.get(error, error))
+
+
+def main():
+    argument_spec = dict(
+        storagebox_id=dict(type='int', required=True),
+        snapshot_name=dict(type='str'),
+        state=dict(type='str', default="present",
+                   choices=['present', 'absent']),
+        snapshot_comment=dict(type='str')
+    )
+
+    argument_spec.update(ROBOT_DEFAULT_ARGUMENT_SPEC)
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_by={"snapshot_comment": ["snapshot_name"]},
+        required_if=[["state", "absent", ["snapshot_name"]]],
+        supports_check_mode=True
+    )
+
+    storagebox_id = module.params['storagebox_id']
+    state = module.params['state']
+    snapshot_name = module.params['snapshot_name']
+    snapshot_comment = module.params['snapshot_comment']
+
+    # Create snapshot
+    if state == 'present' and not snapshot_name:
+        if module.check_mode:
+            module.exit_json(changed=True)
+        snapshot = create_snapshot(module, storagebox_id)
+        module.exit_json(changed=True, snapshot=snapshot)
+
+    # Update snapshot comment
+    elif state == 'present' and snapshot_name:
+        if not snapshot_comment:
+            module.fail_json(
+                msg="snapshot_comment is required when updating a snapshot")
+
+        snapshots = fetch_snapshots(
+            module=module, storagebox_id=storagebox_id)
+        snapshot = get_snapshot_by_name(snapshots, snapshot_name)
+        if not snapshot:
+            handle_errors(module, "SNAPSHOT_NOT_FOUND",
+                          snapshot_name=snapshot_name)
+        if snapshot_comment != snapshot['comment']:
+            if not module.check_mode:
+                update_snapshot_comment(
+                    module, storagebox_id, snapshot_name, snapshot_comment)
+            module.exit_json(changed=True)
+        else:
+            module.exit_json(changed=False)
+
+    # Delete snapshot
+    elif state == 'absent':
+        snapshots = fetch_snapshots(module=module, storagebox_id=storagebox_id)
+        snapshot = get_snapshot_by_name(snapshots, snapshot_name)
+        if snapshot:
+            if not module.check_mode:
+                delete_snapshot(module, storagebox_id, snapshot_name)
+            module.exit_json(changed=True)
+        else:
+            module.exit_json(changed=False)
+
+
+def delete_snapshot(module, storagebox_id, snapshot_name):
+    url = "{0}/storagebox/{1}/snapshot/{2}".format(
+        BASE_URL, storagebox_id, snapshot_name)
+    dummy, error = fetch_url_json(module, url, method="DELETE", accept_errors=[
+        "STORAGEBOX_NOT_FOUND", "SNAPSHOT_NOT_FOUND"], allow_empty_result=True)
+    if error and error == "STORAGEBOX_NOT_FOUND":
```

Comment:
I'd probably write this as
```suggestion
    if error and error != "SNAPSHOT_NOT_FOUND":
```
so that if at some point more errors are added to `accept_errors`, the condition here doesn't need to be adjusted.
Also I think it makes it easier to understand that you only want to report an error if it's not `SNAPSHOT_NOT_FOUND`.